scylla-cdc/src/reader.rs

```diff
@@ -0,0 +1,333 @@
+        sleep(time::Duration::from_secs(SLEEP_INTERVAL)).await;
+        session
+            .query(
+                format!(
+                    "INSERT INTO {} (pk, t, v, s) VALUES (0, 2, 'val2', 'static2');",
+                    TEST_TABLE
+                ),
+                (),
+            )
+            .await?;
+
+        Ok(())
+    }
+
+    async fn get_cdc_stream_id(session: &Arc<Session>) -> Result<StreamID, Box<dyn Error>> {
+        let query_stream_id = format!(
+            "SELECT \"cdc$stream_id\" FROM {}_scylla_cdc_log LIMIT 1;",
+            TEST_TABLE
+        );
+
+        let mut rows = session
+            .query_iter(query_stream_id, &[])
+            .await?
+            .into_typed::<StreamID>();
+
+        Ok(rows.next().await.unwrap()?)
+    }
+
+    async fn delete_test_data_in_db(session: &Session) -> Result<(), Box<dyn Error>> {
+        session
+            .query(format!("DROP KEYSPACE IF EXISTS {}", TEST_KEYSPACE), ())
+            .await?;
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn fetch_cdc_data() -> Result<(), Box<dyn Error>> {
+        let uri = std::env::var("SCYLLA_URI").unwrap_or_else(|_| "172.17.0.2:9042".to_string());
+        let session = SessionBuilder::new().known_node(uri).build().await?;
+        let shared_session = Arc::new(session);
+        let mut cdc_reader = match setup(&shared_session).await {
+            Ok(reader) => reader,
+            Err(e) => {
+                delete_test_data_in_db(&shared_session).await?;
+                return Err(e);
+            }
+        };
+        test_fetch_cdc(&shared_session, &mut cdc_reader).await?;
+        delete_test_data_in_db(&shared_session).await?;
+        Ok(())
+    }
+
+    // #[tokio::test]
+    // async fn fetch_with_set_upper_timestamp() -> Result<(), Box<dyn Error>> {
+    //     let uri = std::env::var("SCYLLA_URI").unwrap_or_else(|_| "172.17.0.2:9042".to_string());
+    //     let session = SessionBuilder::new().known_node(uri).build().await?;
+    //     let shared_session = Arc::new(session);
+    //     let mut cdc_reader = match setup(&shared_session).await {
+    //         Ok(reader) => reader,
+    //         Err(e) => {
+    //             delete_test_data_in_db(&shared_session).await?;
+    //             return Err(e);
+    //         }
+    //     };
+    //     let to_set_upper_timestamp = 2;
+    //     cdc_reader
+    //         .set_upper_timestamp(chrono::Duration::seconds(
+    //             chrono::Local::now().timestamp() - to_set_upper_timestamp,
+    //         ))
+    //         .await;
+    //     test_fetch_cdc(&shared_session, &mut cdc_reader).await?;
+    //     delete_test_data_in_db(&shared_session).await?;
+    //     Ok(())
+    // }
+
+    async fn test_fetch_cdc(
+        shared_session: &Arc<Session>,
+        cdc_reader: &mut StreamReader,
+    ) -> Result<(), Box<dyn Error>> {
+        match cdc_reader
+            .fetch_cdc(TEST_KEYSPACE.parse()?, TEST_TABLE.parse()?)
```

Comment:
Why use `parse()` here?? I think you just want to convert `&str` to `String`? If so, then please just use `to_string()`.